security/sshdo/files/macports.patch

```diff
@@ -0,0 +1,38 @@
+--- Makefile.orig	2023-06-19 15:23:20.000000000 +1000
++++ Makefile	2023-06-19 15:58:11.000000000 +1000
+@@ -58,8 +58,8 @@
+ 	@echo "make diff      - Show differences between source and installed versions"
+ 
+ test:
+-	@[ -z "`which python3 2>/dev/null | grep '^/'`" -a -n "`which python 2>/dev/null | grep '^/'`" ] && sed 's/env python3$$/env python/' < ./sshdo > ./sshdo2 && chmod 755 ./sshdo2 && mv ./sshdo2 ./sshdo || true
+-	@[ -z "`which python3 2>/dev/null | grep '^/'`" -a -n "`which python 2>/dev/null | grep '^/'`" ] && sed 's/env python3$$/env python/' < ./test_sshdo > ./test_sshdo2 && chmod 755 ./test_sshdo2 && mv ./test_sshdo2 ./test_sshdo || true
++	@[ -z "`which python@PYTHON_BRANCH@ 2>/dev/null | grep '^/'`" -a -n "`which python 2>/dev/null | grep '^/'`" ] && sed 's/env python3\.11$$/env python/' < ./sshdo > ./sshdo2 && chmod 755 ./sshdo2 && mv ./sshdo2 ./sshdo || true
++	@[ -z "`which python@PYTHON_BRANCH@ 2>/dev/null | grep '^/'`" -a -n "`which python 2>/dev/null | grep '^/'`" ] && sed 's/env python3\.11$$/env python/' < ./test_sshdo > ./test_sshdo2 && chmod 755 ./test_sshdo2 && mv ./test_sshdo2 ./test_sshdo || true
```

Comment:
Two hardcoded references to `3\.11` remain here (and another below).

You shouldn't need to run `which` on the python binary to see if it exists; you can be certain it exists because the port declares a dependency on it.

`which` isn't the best way to determine if a program exists; `command -v` is.

It looks like the original code in the Makefile is trying to do this: if the user does not have `python3`, but does have `python`, then it replaces `env python3` with `env python` in the sshdo and test_sshdo scripts. For MacPorts, we would always want to replace `/usr/bin/env python3` with the absolute path of the MacPorts python that we're using. For example (I have not tested this):

```suggestion
+	@sed '1s,#!/usr/bin/env python3$$,#!@PREFIX@/bin/python@PYTHON_BRANCH@,' < ./sshdo > ./sshdo2 && chmod 755 ./sshdo2 && mv ./sshdo2 ./sshdo || true
+	@sed '1s,#!/usr/bin/env python3$$,#!@PREFIX@/bin/python@PYTHON_BRANCH@,' < ./test_sshdo > ./test_sshdo2 && chmod 755 ./test_sshdo2 && mv ./test_sshdo2 ./test_sshdo || true
```

However, you're also patching sshdo and test_sshdo later in the patchfile; only one or the other is needed, and patching the shebang line with placeholders and reinplacing them in the portfile is the usual method. In that case:

```suggestion
```

Reply:
Actually, those rewrites can completely disappear for macports because it is known that python311 will be present. I've removed them.